exporter/awsxrayexporter/internal/translator/cause.go

```diff
@@ -130,11 +128,19 @@ func makeCause(span ptrace.Span, attributes map[string]pcommon.Value, resource p
 			if code == 429 {
 				isThrottle = true
 			}
-		} else {
+		} else if code >= 500 && code <= 599 {
 			isError = false
 			isThrottle = false
 			isFault = true
+		} else {
```

Comment:
Should we check span status code before we set `isError=false`?

Reply:
I was thinking to use `http.status_code` alone to determine X-Ray segment status, but your suggestion make sense for the edge cases. We'll still check status_code `400-599` for determine the error/fault segments first, and then use Span status.code() to set `isFault` in X-Ray segment if the status code is true